lambda/checksum.tf

```diff
@@ -78,15 +78,28 @@ resource "aws_security_group" "allow_efs_lambda_checksum" {
   description = "Allow EFS inbound traffic"
   vpc_id      = var.vpc_id
 
-  egress {
-    protocol    = "-1"
-    from_port   = 0
-    to_port     = 0
-    cidr_blocks = ["0.0.0.0/0"]
-  }
-
   tags = merge(
     var.common_tags,
     map("Name", "${var.project}-lambda-allow-efs-checksum-files")
   )
 }
+
+resource "aws_security_group_rule" "allow_https_lambda_checksum_rule" {
+  count             = local.count_checksum
+  from_port         = 443
+  protocol          = "tcp"
+  security_group_id = aws_security_group.allow_efs_lambda_checksum[count.index].id
+  to_port           = 443
+  type              = "egress"
+  cidr_blocks       = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group_rule" "efs_rule" {
```

Comment:
How easy would it be to extract small modules for the HTTPS rule and the EFS rule? They're identical for every Lambda except for the security group, which you could pass in as a parameter.

It might help to keep the Lambda modules short and easy to understand.

Reply:
I don't think it's going to make it much clearer. Each module needs it's own `main.tf` and `variables.tf`. I did try it and the rule declaration in the checksum file now looks like

```
module allow_https_lambda_checksum_rule {
  source = "./modules/https_security_group_rule"
  security_group_id = aws_security_group.allow_efs_lambda_checksum
  rule_count = local.count_checksum // need this as the lambdas are created conditionally
}

module "allow_efs_lambda_checksum_role" {
  source = "modules/efs_security_group_rule"
  rule_count = local.count_checksum
  security_group_id = aws_security_group.allow_efs_lambda_checksum
  source_security_group = var.efs_security_group_id //need this because efs_security_group_id is passed into a variable to the lambda module
}
```

We'll still have all of the duplication we've currently got because the calls to the modules will be identical apart from the security group and the count parameter and we're only saving 3 or 4 lines so it's not going to make it much easier to read. You're right that there is a lot of duplication but it's probably best to keep all the lambda `.tf` files self contained.

Reply:
Fair enough. I guess the fact that each resource has a name like `allow_efs_lambda_download_files_rule` means that its purpose is clear enough as it is.

Though if we ever do want to build a small module, it doesn't have to have `main.tf` and `variables.tf`. That's just a convention, and if a module is only 20 lines long with a single variable then it might be more readable to put the variable into the same file as the resources.

Reply:
Yeah true. I'll leave it as is for the moment then and keep each lambda self contained. Once MVP is done, I would like to have a look at the terraform structure and see if we can clean it up a bit anyway.